src/async_fuse/memfs/s3_metadata.rs

```diff
@@ -305,28 +269,49 @@ impl<S: S3BackEnd + Sync + Send + 'static> MetaData for S3MetaData<S> {
         //         flags,
         //     ))
         // } else {
-        retry_txn!(TXN_RETRY_LIMIT, {
+        // First find in open_files
+        let o_flags = fs_util::parse_oflag(flags);
+        let access_mode = match o_flags & (OFlag::O_RDONLY | OFlag::O_WRONLY | OFlag::O_RDWR) {
+            OFlag::O_RDONLY => 4,
+            OFlag::O_WRONLY => 2,
+            _ => 6,
+        };
+        if let Some(open_file) = self.open_files.try_open(ino) {
+            let open_file = open_file.read();
+            open_file
+                .attr
+                .check_perm(context.user_id, context.group_id, access_mode)?;
+            return Ok(GLOBAL_S3_FD_CNT.fetch_add(1, Ordering::SeqCst).cast());
+        }
+        // The file was not opened before, so we need to open it.
+        let (result, attr) = retry_txn!(TXN_RETRY_LIMIT, {
             let mut txn = self.kv_engine.new_meta_txn().await;
             let node = self.get_inode_from_txn(txn.as_mut(), ino).await?;
-            let o_flags = fs_util::parse_oflag(flags);
-            node.open_pre_check(o_flags, context.user_id, context.group_id)?;
-
+            let attr = node.get_attr();
+            attr.check_perm(context.user_id, context.group_id, access_mode)?;
             let result = node.dup_fd(o_flags).await;
```

Comment:
If we do not maintain the `open_count` in `S3Node` really, we can remove the field from `S3Node`, so that an `open()` call would not need to commit anything to the KV.

Reply:
Removed

Reply:
`node` seems not to be modified, so do we still need to commit the node to KV?

Reply:
I will fix it in this pull request later.